### ISSUE TITLE
'have_output_on_stdout' doesn't work with 'all_commands'

### ISSUE THREAD

**Issue by peagha** [cucumber/aruba#576]
## Summary

I'm fixing the deprecation warnings in my app, and message suggests me to use `expect(all_commands).to have_output_on_stdout /output/`, but the matcher doesn't seem to work with `all_commands `.

## Expected Behavior
Given a command:
```
run('echo output')
```
and the expectation:
```
expect(all_commands).to have_output_on_stdout /output/
```
The test should pass.
## Current Behavior
The test fails with the message:
```
expected [#<Aruba::Command(#<Aruba::Processes::SpawnProcess:70349724675560 commandline="echo output": output="output
">)>] to have output on stdout: /output/
Diff:
@@ -1,2 +1,3 @@
-/output/
+[#<Aruba::Command(#<Aruba::Processes::SpawnProcess:70349724675560 commandline="echo output": output="output
+">)>]
 (RSpec::Expectations::ExpectationNotMetError)
```
## Possible Solution
The `have_output_on_stdout` matcher [expects an object that responds to `stdout`](https://github.com/cucumber/aruba/blob/8b74a97942327067bbf10ae35835615e697fb425/lib/aruba/matchers/command/have_output_on_stdout.rb#L20):
- update `all_commands` to respond to `stdout`
- update `have_output_on_stdout` to handle Arrays
- update the deprecation warning to use `last_command_started`

Maybe the matcher can also be updated to provide a more meaningful message when it receives an object it can't handle.

## Steps to Reproduce (for bugs)
(I didn't provide a list of steps because the bug is fairly easy to reproduce. If you need more info let me know)

## Context & Motivation
I wasn't able to follow the warnings and update my app successfully.
## Your Environment

* Version used: Aruba 0.14.6
* Operating System and version: macOs El Capitan
* Link to your project: N/D

**Comment by peagha** [comment]
`/unstale`

**Comment by mvz** [comment]
Fixed in #601 by changing the deprecation message.
